src/services/professorScore.ts

```diff
@@ -35,3 +40,140 @@ export const upsertProfessorScore = async (
 
   return newScore;
 };
+
+export const computeProfessorScore = async (
+  userId: number,
+): Promise<UpdateProfessorScoreDto> => {
+  const filterActivitesBySigLevel = (
+    activites: Activity[],
+    sigLevel: SignificanceLevel,
+  ) => {
+    return activites.filter((activity) => activity.significance === sigLevel);
+  };
+
+  const filterActivitesForCategory = (
+    activites: Activity[],
+    category: ActivityCategory,
+  ) => {
+    return activites.filter((activity) => activity.category === category);
+  };
+
+  const activites = await getActivitiesForUser(userId);
+  const approvedActivities = activites.filter(
+    (activity) => activity.meritStatus === 'ACCEPTED',
+  );
+
+  let teachingScore = 6;
+  let researchScore = 6;
+  let serviceScore = 6;
+
+  const categories: ActivityCategory[] = ['TEACHING', 'RESEARCH', 'SERVICE'];
+
+  categories.forEach((category) => {
+    const filteredActivities = filterActivitesForCategory(
+      approvedActivities,
+      category,
+    );
+
+    const majorActivities = filterActivitesBySigLevel(
+      filteredActivities,
+      'MAJOR',
+    ).length;
+    const minorActivities = filterActivitesBySigLevel(
+      filteredActivities,
+      'MINOR',
+    ).length;
+    const significantActivites = filterActivitesBySigLevel(
+      filteredActivities,
+      'SIGNIFICANT',
+    ).length;
+
+    if (category === 'TEACHING') {
```

Comment:
couple suggestions to clean up this function:
- rather than having 3 variables for each category and referencing them manually, there are cleaner ways to abstract this logic. we could create a record mapping categories to scores:
   ```
   const categoryScores: Record<ActivityCategory, number> = { 'MINOR': 6, 'MAJOR': 6, 'SIGNIFICANT': 6 };
   ```
   that way we can easily access scores via `categoryScores[category]` rather than checking "if we're dealing with teaching, then modify the teaching variable" for each category

- could we move lines 91-170 into a helper method? this is a pretty complicated function and it would help to separate the scoring logic from the actual service function. I'm thinking we could create a function that scores each category, e.g. `categoryScore(category, numMajor, numMinor, numSignificant): number`. that way we can easily calculate/store the scores for each category without enumerating each case:
   ```
   const score = categoryScore(category, majorActivities, minorActivities, significantActivites);
   categoryScores[category] = score;
   ```